sunpy/net/jsoc/jsoc.py

```diff
@@ -466,23 +466,31 @@ def _make_recordset(self, start_time, end_time, series, wavelength='',
                         segment='', **kwargs):
         # Build the dataset string
         # Extract and format Wavelength
+        
         if wavelength:
             if not series.startswith('aia'):
                 raise TypeError("This series does not support the wavelength attribute.")
             else:
-               if isinstance(wavelength, list):
-                   wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
-                   wavelength = str(wavelength)
-               else:
-                   wavelength = '[{0}]'.format(int(np.ceil(wavelength.to(u.AA).value)))
+                if isinstance(wavelength, list):
+                    wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
+                    wavelength = str(wavelength)
+                else:
+                    wavelength = '[{0}]'.format(int(np.ceil(wavelength.to(u.AA).value)))
 
         # Extract and format segment
         if segment != '':
             segment = '{{{segment}}}'.format(segment=segment)
-
-        dataset = '{series}[{start}-{end}]{wavelength}{segment}'.format(
+        
+        period = ''
+        try :
```

Comment:
the more efficient way of writing this is:

```python
period = kwargs.get('period', '')
if period:
    period = '@{}s'.format(period)
```

that saves on the try except and makes it a little neater (an empty string evaluates to false).